homeassistant/components/device_tracker/asuswrt.py

```diff
@@ -219,7 +220,7 @@ def _get_neigh(self):
         devices = {}
         for device in result:
             mac = device['mac'].upper()
-            devices[mac] = Device(mac, None, None)
```

Comment:
Two points: 
* If `ip neigh` can report ARP cache entries without an IP (otherwise you don't need the `.get(…)` here), but `arp -n` cannot (otherwise you'd need `.get(…)` there), you should reverse the order in which `_get_neigh` and `_get_arp` are called: Since `_get_neigh` is called after `_get_arp`, an entry without an IP may overwrite one with IP coming from `_get_arp`.
* I'm not 100% firm regarding the linux network stack, but don't `ip neigh` and `arp -n` query the exact same database? Namely, the ARP neighborhood maintained by the kernel? Is it useful to query both?

Reply:
@tinloaf Can ```ip neigh``` report entries without IP? I know that MAC can be missing, but it seems like IP is always present, right?

Reply:
@tinloaf The first implementation only used ```ip neigh```. ```arp -n```  seems to have been added to support AP mode. I think you are correct that  ```arp -n``` is enough

Reply:
I don't know, but if it's like you say, then the `.get()` is superfluous in line 223, but there should be a `.get('mac', None)` in line 222.